### PR TITLE
ASB  custom token provider

### PR DESCRIPTION
**Pull request by yvesgoeleven**
Shows how to register a custom token provider (f.e. ADFS integration)

### CODE REVIEW

**Comment by SimonCropp** [comment]
needs a rebase

**Comment by yvesgoeleven** [comment]
@SimonCropp done

**Comment by yvesgoeleven** [comment]
@Particular/docs-maintainers @Particular/azure-service-bus-maintainers this PR is up for review

File: Snippets/ASB/ASB_7/ASB_7.csproj [hunk 1]
```diff
@@ -54,6 +54,8 @@
     </Reference>
     <Reference Include="NServiceBus.Azure.Transports.WindowsAzureServiceBus, Version=7.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\NServiceBus.Azure.Transports.WindowsAzureServiceBus.7.0.0-beta0008\lib\net452\NServiceBus.Azure.Transports.WindowsAzureServiceBus.dll</HintPath>
+    <Reference Include="NServiceBus.Azure.Transports.WindowsAzureServiceBus">
```
Comment (SimonCropp) [inline Snippets/ASB/ASB_7/ASB_7.csproj]: seems nuget corrupted this

Comment (yvesgoeleven) [inline Snippets/ASB/ASB_7/ASB_7.csproj]: again? then it's related to rebase and it does not show conflict

Comment (yvesgoeleven) [inline Snippets/ASB/ASB_7/ASB_7.csproj]: should be fixed now

**Pull request merged**